### ISSUE TITLE
ReplyAction="*"

### ISSUE THREAD

**Issue by kansasdev** [CoreWCF/CoreWCF#1131]
I have found problems with wsdl generation when interface of WCF service contains ReplyAction="*". It's quite common that svcutil generates for some services interfaces in that way:
[ServiceContract(Namespace ="http://ns.org")]
    public interface IService
    {
        [OperationContract(Action="http://ns.org/Action",ReplyAction ="*")]
        string GetData(int value);

        [OperationContract(Action ="http://ns.org/Object",ReplyAction ="*")]
        CompositeType GetObject(CompositeType obj);
    }
If you are trying to port such interface to corewcf and
when you are going to view wsdl of service generated by corewcf you are seeing **no operation** part of generated wsdl
<wsdl:operation name="GetData">
<soap:operation soapAction="http://ns.org/Action" style="document"/>
<wsdl:input>
<soap:body use="literal"/>
</wsdl:input>
<wsdl:output>
<soap:body use="literal"/>
</wsdl:output>
</wsdl:operation>
<wsdl:operation name="GetObject">
<soap:operation soapAction="http://ns.org/Object" style="document"/>
<wsdl:input>
<soap:body use="literal"/>
</wsdl:input>
<wsdl:output>
<soap:body use="literal"/>
</wsdl:output>
</wsdl:operation>

If you remove ReplyAction="*" from operationcontract, below wsdl part is visible and wsdl is a proper service.
Is this a correct behavior?

**Comment by mconnew** [comment]
An action of "*" means it can be any value. For an incoming message, that means it matches any message which doesn't match another operation. Only incoming messages can use "*", and on service side (or callback contract on client), only one operation can use *. For the reply action, only the client can use *. This is because the service side needs to apply an action header to the reply message. The client can say it doesn't care what the reply action is at it uses other methods to know which message is the reply. On transports which associate a reply with a request such as HTTP, it's obvious what the reply message is. For duplex transports, it uses the RelatesTo header to know which message it relates to.  

Basically the end which adds the action header to the message can't have an action of *, but the receiver can.

**Comment by kansasdev** [comment]
Understood, thanks. But for compatibility with classic WCF - if you browse to wsdl produced by contract mentioned above (interface), wsdl is ok.
If you browse to the same wsdl made from above interface and created by CoreWCF, wsdl is inproper.
I have found this behavior when I was porting large wcf service to corewcf. Existing interfaces have to be modified.
